### PR TITLE
add an option to enable MutateKubeAPIServerService

### PR DESCRIPTION
**Pull request by jia-jerry**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/priority critical
/platform alicloud

**What this PR does / why we need it**:
Add an option MutateKubeAPIServerService in controller configuration to determine whether the webhook will change KubeApiServerService ExternalTrafficPolicy to Local.
After change ExternalTrafficPolicy to local in Canary, there happened a strange issue. Some pods which happen ran on a specific node couldn't access in-cluster kubeapiserver's cluster ip. It could be calico network issue but the root cause hasn't spotted yet. However, this issue hasn't happpened when ExternalTrafficPolicy was cluster. To be cautious, I added this option. By default, webhook doens't change the ExternalTrafficPolicy.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Add an option MutateKubeAPIServerService in controller configuration to determine whether the webhook changes KubeApiServerService ExternalTrafficPolicy to Local.
```

### CODE REVIEW

**Review by minchaow — DISMISSED** [review]

File: charts/gardener-extension-provider-alicloud/values.yaml [hunk 1]
```diff
@@ -58,6 +58,7 @@ config:
     storage:
       className: gardener.cloud-fast
       capacity: 25Gi
+#  mutateKubeAPIServerService: true
```
Comment (rfranzke) [inline charts/gardener-extension-provider-alicloud/values.yaml:61]: What about a cleaner structure here, i.e.
```suggestion
# kubeAPIServer:
#   mutateExternalTrafficPolicy: true
```

Comment (jia-jerry) [reply charts/gardener-extension-provider-alicloud/values.yaml]: OK. Also changed the config struct.

File: charts/gardener-extension-provider-alicloud/values.yaml [hunk 2]
```diff
@@ -58,6 +58,8 @@ config:
     storage:
       className: gardener.cloud-fast
       capacity: 25Gi
+  kubeApiServer:
```
Comment (rfranzke) [inline charts/gardener-extension-provider-alicloud/values.yaml:61]: nit:

```suggestion
  kubeAPIServer:
```

File: pkg/apis/config/types.go [hunk 3]
```diff
@@ -37,12 +37,20 @@ type ControllerConfiguration struct {
 	MachineImageOwnerSecretRef *corev1.SecretReference
 	// WhitelistedImageIDs specifies an array of image IDs that will bypass image sharing.
 	WhitelistedImageIDs []string
+	//KubeAPIServer is the KubeAPIServer configuration.
```
Comment (rfranzke) [inline pkg/apis/config/types.go:40]:
```suggestion
	// KubeAPIServer is the KubeAPIServer configuration.
```

File: pkg/apis/config/types.go [hunk 4]
```diff
@@ -37,12 +37,20 @@ type ControllerConfiguration struct {
...
+	KubeAPIServer KubeAPIServer
```
Comment (rfranzke) [inline pkg/apis/config/types.go:41]:
```suggestion
	KubeAPIServer *KubeAPIServer
```

File: pkg/apis/config/v1alpha1/types.go [hunk 5]
```diff
@@ -39,13 +39,21 @@ type ControllerConfiguration struct {
 	MachineImageOwnerSecretRef *corev1.SecretReference `json:"machineImageOwnerSecretRef,omitempty"`
 	// WhitelistedImageIDs specifies an array of image IDs that will bypass image sharing.
 	WhitelistedImageIDs []string `json:"whitelistedImageIDs,omitempty"`
+	//KubeAPIServer is the KubeAPIServer configuration.
+	KubeAPIServer KubeAPIServer `json:"kubeAPIServer"`
```
Comment (rfranzke) [inline pkg/apis/config/v1alpha1/types.go:43]:
```suggestion
	// KubeAPIServer is the KubeAPIServer configuration.
	// +optional
	KubeAPIServer *KubeAPIServer `json:"kubeAPIServer,omitempty"`
```

File: pkg/webhook/controlplaneexposure/add.go [hunk 6]
```diff
@@ -38,6 +38,8 @@ var (
 type AddOptions struct {
 	// ETCDStorage is the etcd storage configuration.
 	ETCDStorage config.ETCDStorage
+	//KubeAPIServer is the KubeAPIServer configuration.
```
Comment (rfranzke) [inline pkg/webhook/controlplaneexposure/add.go:41]:
```suggestion
	// KubeAPIServer is the KubeAPIServer configuration.
```

File: charts/gardener-extension-provider-alicloud/templates/configmap.yaml [hunk 7]
```diff
@@ -22,6 +22,8 @@ data:
       name: {{ .Values.config.machineImageOwnerSecret.name }}
       namespace: {{ .Release.Namespace }}
 {{- end }}
+    kubeAPIServer:
+      mutateExternalTrafficPolicy: {{ .Values.config.kubeApiServer.mutateExternalTrafficPolicy }}
```
Comment (rfranzke) [inline charts/gardener-extension-provider-alicloud/templates/configmap.yaml:26]:
```suggestion
      mutateExternalTrafficPolicy: {{ .Values.config.kubeAPIServer.mutateExternalTrafficPolicy }}
```

**Review by rfranzke — COMMENTED** [review]
nits

**Review by rfranzke — COMMENTED** [review]
Thanks, looks great now. The only left suggestion I have is to enhance the example component config file: https://github.com/gardener/gardener-extension-provider-alicloud/blob/master/example/00-componentconfig.yaml

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**